crates/sdl-encoder/src/field_type.rs

```diff
@@ -0,0 +1,65 @@
+use std::fmt::{self, Display};
+
+/// Define Field Type.
+#[derive(Debug, PartialEq, Clone)]
+pub enum FieldType {
+    /// The list field type.
+    List {
+        /// List field type.
+        ty: Box<FieldType>,
+        /// Nullable list.
+        is_nullable: bool,
```

Comment:
I think it may make sense to follow `graphql-js` here and make `NonNull` a type wrapper similar to `List`.

Reply:
That's an interesting idea, let me think of a good way of making it work here.

Reply:
`NonNull` type worked really nicely, thanks!